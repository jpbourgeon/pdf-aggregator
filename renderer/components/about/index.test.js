const React = require('react');
const ShallowRenderer = require('react-test-renderer/shallow');
const { AboutRenderView } = require('../about');

describe('The renderView component of the about modal', () => {
  it('should render correctly', () => {
    const props = {
      aboutState: {
        open: false,
        data: {
          name: 'name',
          description: 'description',
          version: 'version',
          releases: 'releases',
          github: 'github',
          author: 'author',
          email: 'email',
          license: 'license',
        },
      },
      aboutActions: {
        open: jest.fn(),
        close: jest.fn(),
      },
      classes: {
        paper: {},
        cardActions: {},
      },
    };
    const renderer = new ShallowRenderer();
    renderer.render(<AboutRenderView {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
