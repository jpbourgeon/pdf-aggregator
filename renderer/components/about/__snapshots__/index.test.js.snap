// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The renderView component of the about modal should render correctly 1`] = `
<React.Fragment>
  <WithStyles(Modal)
    aria-describedby="simple-modal-description"
    aria-labelledby="simple-modal-title"
    onClose={[MockFunction]}
    open={false}
  >
    <WithStyles(Paper)
      className={Object {}}
    >
      <WithStyles(CardContent)>
        <WithStyles(Typography)
          variant="headline"
        >
          undefined undefined
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="subheading"
        />
        <WithStyles(Typography)
          variant="body1"
        >
          <br />
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="body1"
        >
          <a
            rel="noopener noreferrer"
            target="_blank"
          />
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="body1"
        >
          <a
            rel="noopener noreferrer"
            target="_blank"
          />
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="body1"
        >
          <a
            rel="noopener noreferrer"
            target="_blank"
          />
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="body1"
        >
          <br />
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="body1"
        >
          <a
            rel="noopener noreferrer"
            target="_blank"
          />
           | 
          <a
            rel="noopener noreferrer"
            target="_blank"
          />
        </WithStyles(Typography)>
        <div>
          <WithStyles(Button)
            onClick={[MockFunction]}
            size="small"
            variant="raised"
          />
        </div>
      </WithStyles(CardContent)>
    </WithStyles(Paper)>
  </WithStyles(Modal)>
</React.Fragment>
`;
